File: ReproducerApp/ios/TestEmitter.m
#import "TestEmitter.h"
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(TestEmitter, RCTEventEmitter)
RCT_EXTERN_METHOD(sendEventToJS)
@end

File: ReproducerApp/ios/TestEmitter.swift
//
//  TestEmitter.swift
//  ReproducerApp
//
//  Created by Fabio Pelizzola on 16/01/25.
//

import Foundation

@objc(TestEmitter)
class TestEmitter: RCTEventEmitter {
  
  public override static func moduleName() -> String! {
    return "TestEmitter"
  }
    
    override static func requiresMainQueueSetup() -> Bool {
        return true
    }

    @objc override func supportedEvents() -> [String] {
        return [
            "TEST"
        ]
    }

    @objc func sendEventToJS() {
        self.sendEvent(withName: "TEST", body: [])
    }
}
